WordPress/Classes/Services/AccountService.m

```diff
@@ -287,4 +287,24 @@ - (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDet
     [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
 }
 
+- (void)purgeSelfHostedAccount:(WPAccount *)account
+{
+    if (!account.isWpcom && [account.blogs count] == 0) {
+        [self.managedObjectContext deleteObject:account];
```

Comment:
This is probably never going to break, depending on how we call the code, but it'd be slightly safer to use the object's MOC when one is available (because there's no chance for the caller to pass an object that's coming from a different MOC or even call this method from a different thread):

```objective-c
[account.managedObjectContext deleteObject:account];
```

The service's would at that point be used for new fetch requests and such.